syfertext/__init__.py

```diff
@@ -97,3 +113,66 @@ def create(pipeline_name, tags: Set[str] = None, description: str = None):
 
 # Set the default owners of some classes
 # SubPipeline.owner = LOCAL_WORKER
+
+
+def save(pipeline_name: str, pipeline: "Pipeline", destination: Union["local"] = "local") -> None:
+    """Saves the pipeline and it's states to storage
+
+    Args:
+        pipeline_name (str): The name of the pipeline.
+        pipeline (Pipeline): The pipeline object itself
+        destination (Union): The location where to save that object, currently storage on local machine is implemented.
+
+    """
+
+    # Path to the home/SyferText directory
+    data_path = os.path.join(str(Path.home()), "SyferText")
+
+    # Creating a new directory if home/SyferText does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
+    data_path = os.path.join(data_path, "cache")
+
+    # Creating a new directory if home/SyferText/cache does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
+    data_path = os.path.join(data_path, pipeline_name)
+
+    # Creating a new directory if home/SyferText/cache/<pipeline_name> does not exist
+    if not os.path.exists(data_path):
+        os.mkdir(data_path)
+
+    # Making a target at the file
+    target = str("/{}.pkl".format(pipeline_name))
+
+    # Opening cache file
+    pipeline_cache = open(data_path + target, "wb")
+
+    # Dumping data
+    dill.dump(pipeline.simplify(worker=LOCAL_WORKER, pipeline=pipeline), pipeline_cache)
+
+    # Loading states
+    for state in pipeline.states_info:
+        # Searching on state
+        state_id = create_state_query(pipeline_name=pipeline_name, state_name=state)
+
+        result = search_resource(query=state_id, local_worker=LOCAL_WORKER)
+
+        if isinstance(result, StatePointer):
+            # State target
+            # Since : is a reserved character for naming files, replacing with - instead
+            tokens = state_id.split(":")
+            print("Tokens", tokens)
+            file_name = "-".join(tokens)
```

Comment:
```suggestion
            state_id = state_id.replace(":", "-")
           
```

also no need to print